test/auth/auth.e2e-spec.ts

```diff
@@ -68,6 +71,87 @@ describe('AuthController (e2e)', () => {
     });
   });
 
+  describe('login the registered user', () => {
+    const loginInput: LoginInput = {
+      username: 'e2etester@random.com',
+      password: '12345678',
+    };
+
+    it('should succesfully login the user', () => {
+      return request(app.getHttpServer())
+        .post('/auth/login')
+        .send(loginInput)
+        .expect(HttpStatus.OK)
+        .expect((res) => {
+          const token = res.body;
+          expect(token).toHaveProperty('accessToken');
+          expect(token).toHaveProperty('refreshToken');
+        });
+    });
+
+    it('should failed to login with wrong credential', () => {
+      return request(app.getHttpServer())
+        .post('/auth/login')
+        .send({ ...loginInput, password: 'wrong-passs' })
+        .expect(HttpStatus.UNAUTHORIZED);
+    });
+  });
+
+  describe('login the registered user', () => {
+    const loginInput: LoginInput = {
+      username: 'e2etester@random.com',
+      password: '12345678',
+    };
+
+    it('should succesfully login the user', () => {
+      return request(app.getHttpServer())
+        .post('/auth/login')
+        .send(loginInput)
+        .expect(HttpStatus.OK)
+        .expect((res) => {
+          const token = res.body;
+          expect(token).toHaveProperty('accessToken');
+          expect(token).toHaveProperty('refreshToken');
+        });
+    });
+
+    it('should failed to login with wrong credential', () => {
+      return request(app.getHttpServer())
+        .post('/auth/login')
+        .send({ ...loginInput, password: 'wrong-passs' })
+        .expect(HttpStatus.UNAUTHORIZED);
+    });
+  });
+
+  describe('refreshing jwt token', () => {
+    const loginInput: LoginInput = {
+      username: 'e2etester@random.com',
+      password: '12345678',
+    };
+
+    it('should succesfully get new auth token using refresh token', () => {
+      return request(app.getHttpServer())
+        .post('/auth/login')
+        .send(loginInput)
+        .then((res) => {
+          const token: AuthTokenOutput = res.body;
+          const refreshTokenInput: RefreshTokenInput = {
+            refreshToken: token.refreshToken,
+          };
+
+          return request(app.getHttpServer())
```

Comment:
Shall we move this out of the previous `request`'s scope?
We can assign the returned res to `refreshTokenInput` var declared outside this request.

It will make it cleaner for us to read it as well, right now it looks like it's going towards call back hell! :bow:

Reply:
used await the login call, not sure whether this is what you wanted.